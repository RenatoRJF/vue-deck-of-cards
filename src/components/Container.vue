<template>
  <div class="app__container">
    <header>
      <h1>{{ title }}</h1>
    </header>

    <slot />
  </div>
</template>

<script lang="ts">
import { Vue, Component, Prop } from "vue-property-decorator";

@Component
export default class Container extends Vue {
  @Prop() title!: string;
}
</script>

<style lang="scss" scoped>
.app__container {
  max-width: 1366px;
  margin: 0 auto;
  background-color: #dff4f7;

  header {
    background-color: #1e7500;

    h1 {
      color: #fae500;
      font-size: 30px;
      padding: 20px 0;
      text-align: center;
      text-transform: uppercase;
    }
  }
}
</style>
