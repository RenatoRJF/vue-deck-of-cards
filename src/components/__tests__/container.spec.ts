import { shallowMount } from "@vue/test-utils";
import Container from "@/components/Container.vue";

describe("Container.vue", () => {
  it("renders props.title when passed", () => {
    const title = "Cards";
    const wrapper = shallowMount(Container, {
      propsData: { title },
    });
    expect(wrapper.text()).toMatch(title);
    expect(wrapper).toMatchSnapshot();
  });
});
