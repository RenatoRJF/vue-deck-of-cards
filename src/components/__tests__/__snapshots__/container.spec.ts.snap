// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container.vue renders props.title when passed 1`] = `
<div class="app__container">
  <header>
    <h1>Cards</h1>
  </header>
</div>
`;
